tutorials/cloud-run-with-pomerium-for-end-user-access/index.md

```diff
@@ -0,0 +1,190 @@
+route. This turns Pomerium into a bridge between a user-centric and a service-centric authorization model.
+
+## Setup
+
+To deploy Pomerium to Cloud Run, a [special image](https://console.cloud.google.com/gcr/images/pomerium-io/GLOBAL/pomerium) is available at
+`gcr.io/pomerium-io/pomerium-[VERSION]-cloudrun`. It allows sourcing configuration from Google Cloud Secret Manager, and it sets some defaults for Cloud Run to 
+keep configuration minimal. This example uses it to store identity provider credentials. Pomerium's
+[authorization policy](https://www.pomerium.com/reference/#policy) contains no secrets, so you can place it directly in an
+[environment variable](https://www.pomerium.io/reference/#configuration-settings).
+
+The [Dockerfile for this setup for Cloud Run](https://github.com/pomerium/pomerium/blob/master/.github/Dockerfile-cloudrun) is based on the Dockerfile in
+[vals-entrypoint](https://github.com/pomerium/vals-entrypoint).
+
+This image depends on a configuration file present at `/pomerium/config.yaml`, and can source it from Google Cloud Secret Manager. To do so, set
+`VALS_FILES=[secretref]:/pomerium/config.yaml` and set any other [Pomerium environment variables](https://www.pomerium.io/reference/#configuration-settings) 
+directly or with an additional `secretref`.
+
+The `secretref` format for Google Cloud Secret Manager is `ref+gcpsecrets://PROJECT/SECRET(#/key])`.
+
+### Identity provider credentials and secrets in `config.yaml`
+
+Set up Pomerium's [`config.yaml`](https://www.pomerium.com/reference/#shared-settings) to contain your identity provider credentials and secrets:
+
+    # config.yaml
+    authenticate_service_url: https://authn.cloudrun.pomerium.com
+    shared_secret: XXXXXX
+    cookie_secret: XXXXXX
+    idp_provider: "google"
+    idp_client_id: XXXXXX
+    idp_client_secret: "XXXXXX"
+
+### Subdomain and e-mail domain in `policy.template.yaml`
+
+Substitute `cloudrun.pomerium.com` for your own subdomain and your e-mail domain if appropriate:
+
+    # policy.template.yaml
+    # see https://www.pomerium.com/reference/#policy
+    - from: https://hello.cloudrun.pomerium.com
+      to: ${HELLO_URL}
+      allowed_domains:
+        - gmail.com
+      enable_google_cloud_serverless_authentication: true
+    - from: https://httpbin.cloudrun.pomerium.com
+      to: https://httpbin.org
+      pass_identity_headers: true
+      allowed_domains:
+        - gmail.com
+
+### DNS in `zonefile.txt`
+
+Substitute `cloudrun.pomerium.com` for your own subdomain (`zonefile.txt`):
+
+    ; zonefile.txt
+    *.cloudrun.pomerium.com. 18000 IN CNAME ghs.googlehosted.com.
+
+Alternatively, you can set an equivalent CNAME in your DNS provider.
+
+## Deployment
+
+Ensure that you have set a default project:
+
+    gcloud config set default-project MYTESTPROJECT
+
+Run the following commands to configure and deploy Pomerium:
+
+    #!/bin/bash
```

Comment:
the shebang doesn't make a lot of sense here (since we don't tell user to put this into a file) 

we can probably split these into separate code blocks like:

```
    ```
	cmd1
    ```

    ```
	cmd2
    ```
```

Reply:
done